### PR TITLE
Added function for fast-moving block down. It works, but throws a lot of errors.

### PR DESCRIPTION
**Pull request by Arrinao**
Curiously enough, if I use self.canvas which is a correct syntax, the whole thing crashes.

### CODE REVIEW

File: tetris.py [hunk 1]
```diff
@@ -309,6 +322,14 @@ def move_block_right(self, event):
         self.main_board.user_input_right()
         self.main_board.draw_block()
 
+    def move_block_down(self, event):
+        if self.main_board.moving_down is False:
+            self.main_board.moving_down = True
+        else:
+            self.main_board.moving_down = False
```
Comment (Akuli) [inline tetris.py:329]:
```suggestion
        if not self.main_board.moving_down:
            return
```

File: tetris.py [hunk 2]
```diff
@@ -113,6 +115,7 @@ def __init__(self, canvas, width, height, outline_color, current_block_center, s
         self.small_board = small_board
         self.draw_board()
         self.draw_block()
+        self.moving_down = False
```
Comment (Akuli) [inline tetris.py:118]: Why is this in `Board`?

File: tetris.py [hunk 3]
```diff
@@ -256,6 +259,16 @@ def user_input_right(self):
         x, y = self.current_block_center
         self.current_block_center = (x + 1, y)
 
+    def user_input_down(self):
+        if any((x, y + 1) in self.coord_extractor() for (x, y) in self.get_block_shape()) or any(
+            y + 1 == game_height for (x, y) in self.get_block_shape()
+        ):
+            return
+        while self.moving_down is True:
```
Comment (Akuli) [inline tetris.py:267]: `while` loop not necessary, just use `.after()`

**Review by Akuli — CHANGES_REQUESTED** [review]

File: tetris.py (hunk 2 above)
Comment (Arrinao) [reply tetris.py:118]: It's a flag that determines whether the loop in user_input_down should be active or not. Functions move_block_down_press and move_block_down_release affect its state

File: tetris.py (hunk 1 above)
Comment (Arrinao) [reply tetris.py]: Done differently, check out the latest push

File: tetris.py [hunk 4]
```diff
@@ -229,8 +232,9 @@ def current_block_mover(self):
             self.full_line_clear()
             self.new_block()
         else:
-            x, y = self.current_block_center
-            self.current_block_center = (x, y + 1)
+            if self.moving_down is False:
```
Comment (Akuli) [inline tetris.py:235]:
```suggestion
            if not self.moving_down:
```

File: tetris.py [hunk 5]
```diff
@@ -256,6 +260,17 @@ def user_input_right(self):
         x, y = self.current_block_center
         self.current_block_center = (x + 1, y)
 
+    def user_input_down(self):
+        if any((x, y + 1) in self.coord_extractor() for (x, y) in self.get_block_shape()) or any(
+            y + 1 == game_height for (x, y) in self.get_block_shape()
+        ):
+            return
+        if self.moving_down is True:
```
Comment (Akuli) [inline tetris.py:268]:
```suggestion
        if self.moving_down:
```

File: tetris.py [hunk 6]
```diff
@@ -309,6 +324,15 @@ def move_block_right(self, event):
         self.main_board.user_input_right()
         self.main_board.draw_block()
 
+    def move_block_down_press(self, event):
+        if self.main_board.moving_down is True:
```
Comment (Akuli) [inline tetris.py:328]:
```suggestion
        if self.main_board.moving_down:
```

**Review by Akuli — CHANGES_REQUESTED** [review]
Code looks good overall and works well!

When checking for booleans, write `if thing` instead of `if thing is True`, just like you would write `if a == b` instead of `if (a == b) is True`. Also, use `not` instead of `is False`.

**Pull request merged**